controller/src/main/java/io/pravega/controller/store/kvtable/InMemoryKVTMetadataStore.java

```diff
@@ -57,20 +51,12 @@ KeyValueTable newKeyValueTable(String scope, String kvTableName) {
 
     @Override
     public KeyValueTable getKVTable(String scope, final String name, KVTOperationContext context) {
-        KeyValueTable kvt;
```

Comment:
why is this being changed? what was wrong with taking the kvt from the context?

Reply:
There is no advantage in taking KVT from context in case of InMemoryStore as all data is already in memory.
When using KVT from context, the KVTable object is available in the context though it is removed from the scope. Besides it has State="Deleting" which causes the isDeleted() method to never compete.

Reply:
what is the delete method doing? perhaps it should be implemented such that it marks the entry as deleted and removes the entry from the map. 

that way context will be able to get the correct value as well.

Reply:
It should all fall in place once Scopes are implemented correctly in InMemoryKVTMetadataStore.